com.unity.render-pipelines.high-definition/Editor/VFXGraph/VFXHDRPBinder.cs

```diff
@@ -20,52 +20,15 @@ class VFXHDRPBinder : VFXSRPBinder
         public override string SRPAssetTypeStr  { get { return typeof(HDRenderPipelineAsset).Name; } }
         public override Type SRPOutputDataType  { get { return typeof(VFXHDRPSubOutput); } }
 
-        static Dictionary<HDShaderUtils.ShaderID, string> s_ShaderNames = new Dictionary<HDShaderUtils.ShaderID, string>()
-        {
-            { HDShaderUtils.ShaderID.SG_Lit,      "Lit"      },
-            { HDShaderUtils.ShaderID.SG_StackLit, "StackLit" },
-            { HDShaderUtils.ShaderID.SG_Hair,     "Hair"     },
-            { HDShaderUtils.ShaderID.SG_Eye,      "Eye"      },
-            { HDShaderUtils.ShaderID.SG_Fabric,   "Fabric"   },
-            { HDShaderUtils.ShaderID.SG_Unlit,    "Unlit"    },
-        };
-
-        HDShaderUtils.ShaderID GetShaderEnumFromShaderGraph(ShaderGraphVfxAsset shaderGraph)
-        {
-            bool TryGetHDMetadata(out HDMetadata obj)
-            {
-                obj = null;
-
-                var path = AssetDatabase.GetAssetPath(shaderGraph);
-                foreach (var asset in AssetDatabase.LoadAllAssetsAtPath(path))
-                {
-                    if (asset is HDMetadata metadataAsset)
-                    {
-                        obj = metadataAsset;
-                        return true;
-                    }
-                }
-
-                return false;
-            }
-
-            HDMetadata obj;
-
-            if (!TryGetHDMetadata(out obj))
-                throw new ArgumentException("Unknown shader");
-
-            return obj.shaderID;
-        }
-
         public override void SetupMaterial(Material mat, bool hasMotionVector = false, bool hasShadowCasting = false, ShaderGraphVfxAsset shaderGraph = null)
         {
             try
             {
                 if (shaderGraph != null)
                 {
-                    // Recover the HDRP Shader Enum from the VFX Shader Graph.
-                    var shaderID = GetShaderEnumFromShaderGraph(shaderGraph);
-                    HDShaderUtils.ResetMaterialKeywords(mat, shaderID);
+                    // The following will throw an exception if the material with VFX Shader Graph actually doesn't reference an HDRP ShaderGraph (with HDMetaData):
+                    (HDShaderUtils.ShaderID shaderID, GUID subTargetGUID) = HDShaderUtils.GetShaderIDsFromHDMetadata(shaderGraph);
+                    HDShaderUtils.ResetMaterialKeywords(mat);
```

Comment:
@sebastienlagarde @johnpars Note that I've left the old behavior intact, that is to actually throw an exception if the shadergraph asset doesn't have the HDMetaData SO object present, It's not immediately obvious if that was required, but there is a difference so I've kept that subtlety in: if you only call the reset material keywords function, it could silently pass if the shader was not an HD shadergraph. I haven't digged further to see if this can actually happen or if necessary, just providing the same behavior.